src/Logging/Logging.AzureAppServices/src/Internal/BatchingLoggerProvider.cs

```diff
@@ -91,8 +103,10 @@ private async Task ProcessLogQueue(object state)
 
                     _currentBatch.Clear();
                 }
-
-                await IntervalAsync(_interval, _cancellationTokenSource.Token);
+                else
+                {
+                    await IntervalAsync(_interval, _cancellationTokenSource.Token);
```

Comment:
I'm not sure moving this into an else is worth it in a patch. It doesn't seem necessary for dropping messages, and it has the potential to make ProcessLogQueue more CPU hungry. Unless we have some benchmark or stress scenario that shows a major benefit.

Reply:
It allows dropping fewer messages by unloading them faster. Without doing this and having no backpressure a lot of messages might be lost even if app is not overloaded